lib/javascripts/base_app.js

```diff
@@ -55,17 +48,26 @@ function BaseApp(zafClient, data) {
   this.I18n = { t: I18n.t };
   registerHelpers(this);
   bindEvents(this);
-  let evt = { firstLoad: true };
+
   this._metadata = data.metadata;
   this._context = data.context;
-  zafClient.get('currentUser.locale').then(function(data) {
-    I18n.loadTranslations(data['currentUser.locale']);
-    if (this.defaultState) {
-      this.switchTo(this.defaultState);
-    }
-    resolveHandler(this, 'app.created')();
-    resolveHandler(this, 'app.activated')(evt, evt);
-  }.bind(this));
+  this._storage = new Storage(this._metadata.installationId);
+
+  let view = new View({ onRender: () => {
```

Comment:
> Seems overkill to require subclassing just to define an `onRender` hook

Seems awkward to have a thing that is _like_ a mixin, but passes methods in via a constructor as options.